src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,124 @@
 package core.basesyntax;
 
+import java.lang.reflect.Array;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_SIZE = 10;
+    private static final double GROW_FACTOR = 1.5;
+
+    private T[] elements;
+    private int size;
+
     @Override
     public void add(T value) {
-
+        addCellForNewElement(value);
+        elements[size - 1] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        int prevSize = size;
+        addCellForNewElement(value);
+        validateIndex(index);
+        if (index < prevSize) {
+            System.arraycopy(elements, index, elements, index + 1, prevSize - index);
+        }
+        elements[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list == null) {
+            throw new NullPointerException("Input list is null");
+        }
+        if (list.size() == 0) {
+            return;
+        }
```

Comment:
Code makes no impact, if list size is 0 then next cycle is not triggered and we have same result
```suggestion
```

Reply:
Right. Thank you!